base_delivery_carrier_files/models/stock_move.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2012 Camptocamp SA
+# Author: Guewen Baconnier
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def write(self, values):
+        write_result = super(StockMove, self).write(values)
+        if values.get("state") == "done":
+            done_pickings = self.mapped("picking_id").filtered(
+                lambda sp: sp.state == "done"
+            )
```

Comment:
May I know the benefit of using filtered+lambda over search+domain? Thanks

Reply:
Example: https://github.com/odoo/odoo/blob/c08082d89782b7675e8e649cba15ca9e03e40077/addons/purchase_stock/tests/test_replenish_wizard.py#L51

Reply:
Good question! Because the processes of pickings and their stock moves are very much intertwined, it is likely that at this point we will already have the picking states in our cache so then we avoid the search as well as a potential read. Also, when setting the state of more than one stock move, it is unlikely that there are a large number of pickings involved. I would not prefer to use `filtered` if the result set was potentially large (but it also depends on whether the column used in the domain is indexed, which happens to be the case here).